t5x/contrib/gpu/scripts_gpu/README.md

```diff
@@ -0,0 +1,79 @@
+# GPU Scripts
+This directory contains scripts optimized for GPU usage. 
+
+Install with `pip install -r pile_requirements.txt` to get all pile dependencies.
+
+## Building the container
+The Dockerfile at the top level given will build a container with all gpu/pile dependencies. It can be built with `cd ../../; docker/build.sh` 
+
+## Running interactively
+Note: this should only be done with singlenode jobs and/or for downloading the pile. Use `docker/interactive_pull_and_launch.sh`. This takes arguments for the URL to pull a container from and the location of the dataset directory to mount. For example:
+
+`cd ../../; docker/interactive_pull_and_launch.sh [URL] /my/dataset/dir`
+
+## Downloading The Pile
+Run `download_the_pile.py` to download the pile. It will download to the directory set in the environment variable: `TFDS_DATA_DIR`. After that, set the `TFDS_DATA_DIR` to the same directory in your scripts to use.
+
+## Single Node runs
+Pretraining and Finetuning can be done with `singlenode_*.sh`. These will build a T5X model with the Adam optimizer and relevant parameters. These will allow multi-gpu on one host.
+
+## Multi Node runs
+For a SLURM+pyxis cluster, `example*.sub` files provide example slurm submit files (edit with your details), which call `multiprocess*.sh` to execute training. You can add a binding script in the `.sub` file for your cluster, or remove it entirely (dropping some throughput)
+
+## Convergence
+For our Pile convergence runs, we used a Global batch size of `2048` for all models, where GBS is defined as `#GPUs * BS/GPU / Model_Parallel`. Below are example (tested) hardware topologies on NVIDIA DGX A100 (8x A100 80G) nodes.
+
+| size | #GPUs | BS / GPU | Sequences/Sec | Estimated Walltime | MNLI 2.0 - matched | SQuAD v1.1 (EM/F1) | Convergence Log | 
```

Comment:
Hi @SahilJain314! I am currently using t5x on GPUs as well and these benchmarks are super helpful; thanks for contributing this. Do you mind adding the model size in number of parameters to this chart? `t5x/contrib/gpu/t5/t5_1_1/small.gin` indicates that it's a different architecture than what's known as "t5-small" on the [Hugging Face Hub](https://huggingface.co/t5-small/blob/main/config.json) so it took me a bit of digging to figure out how big "small" was :)

Reply:
Hey if you run any job, it will write out `model-info.txt` file to the logdir. That should have a detailed information of the parameters.

FYI t5.1.1 is a slightly different (and improved) architecture than t5-small (the ones introduced in the t5 paper).